src/emulator/modules/SceTouch/SceTouch.cpp

```diff
@@ -55,16 +137,22 @@ static int peek_touch(HostState &host, SceUInt32 port, SceTouchData *pData, SceU
 
         if ((touch_pos_viewport.x >= 0) && (touch_pos_viewport.y >= 0) && (touch_pos_viewport.x < 1) && (touch_pos_viewport.y < 1)) {
             pData->report[pData->reportNum].x = static_cast<uint16_t>(touch_pos_viewport.x * 1920);
-            pData->report[pData->reportNum].y = static_cast<uint16_t>(touch_pos_viewport.y * 1088);
+            if (port == SCE_TOUCH_PORT_FRONT) {
+                pData->report[pData->reportNum].y = static_cast<uint16_t>(touch_pos_viewport.y * 1088);
+            } else {
+                pData->report[pData->reportNum].y = static_cast<uint16_t>(108 + touch_pos_viewport.y * (889 - 108));
+            }
             ++pData->reportNum;
         }
 
         if (!host.ctrl.touch_mode[port]) {
             pData->reportNum = 0;
         }
+    } else if (registered_touch() == true && port == touchscreen_port) {
+        pData[0] = recover_touch_events();
```

Comment:
i would prefer this function to have a SceTouchData* as argument instead of returning a SceTouchData. But not really required.